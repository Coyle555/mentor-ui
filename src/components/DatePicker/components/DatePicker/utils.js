export const isCallbackValid = (callback) => {
	return typeof callback === 'function';
};
