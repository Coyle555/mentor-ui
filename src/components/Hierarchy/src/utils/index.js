export * from './appendNodes';
export * from './collapseNode';
export * from './convertTree';
export * from './expandNode';
export * from './findNode';
