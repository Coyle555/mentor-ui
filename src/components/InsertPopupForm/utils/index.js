export * from './getInputComponent';
export * from './sortFormFields';
