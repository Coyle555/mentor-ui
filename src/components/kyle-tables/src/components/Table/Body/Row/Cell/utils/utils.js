import moment from 'moment';

const datetimeFormat = 'MMM DD, YYYY, h:mm A';
const dateFormat = 'MMM DD, YYYY';

function convertDateTime(datetimeValue, isUtc) {
	console.log('datetime vlaue', datetimeValue);
	return isUtc
		? moment.utc(datetimeValue).local().format(datetimeFormat)
		: moment(datetimeValue, moment.ISO_8601).format(datetimeFormat);
}

function convertDate(dateValue, isUtc) {
	return isUtc
		? moment.utc(dateValue).local().format(dateFormat)
		: moment(dateValue, moment.ISO_8601).format(dateFormat);
}

// Converts a cell value to a string
//
// @value(string|object|null|[object]) - the value passed into a cell
// @return(string) - the value in string format
export function convertCellToString(value, type, isUtc) {
	if (value === null || value === undefined) {
		return '';
	}

	const valueType = typeof value;

	if (type === 'datetime' && Date.parse(value)) {

		return convertDateTime(value, isUtc);

	} else if (type === 'date' && Date.parse(value)) {

		return convertDate(value, isUtc);

	// handle primitive data types
	} else if (valueType === 'string' || valueType === 'number' || valueType === 'boolean') {

		return value.toString();


	// default to name field for objects
	} else if (valueType === 'object' && value.name) {

		return value.name;

	}

	return '';
};
