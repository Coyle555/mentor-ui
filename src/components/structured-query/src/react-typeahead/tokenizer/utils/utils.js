import {
	ALL_OPERATIONS,
	ASYNC_OPERATIONS,
	ENUM_OPERATIONS,
	NUM_DATE_OPERATIONS,
	STRING_OPERATIONS
} from '../constants';

// make sure a token is valid before the structured filter uses it
export function validateToken(token) {
	if (typeof token !== 'object'
		|| !token.id
		|| !token.category
		|| !token.operator) {

		return false;
	}

	// no valid value when operation isnt a empty check
	if (!token.value
		&& token.operator !== ALL_OPERATIONS.IS_EMPTY
		&& token.operator !== ALL_OPERATIONS.IS_NOT_EMPTY) {

		return false;
	}

	// operation does not exist in list of approved operations
	if (!ALL_OPERATIONS[token.operator.toUpperCase().replace(/ /g, '_')]) {
		return false;
	}

	return true;
}

// Get the options available based on where the user is in the query
export function _getOptionsForTypeahead(options = [], token = {}) {
	if (!token.category) {

		return options.map(option => option.category);

	} else if (!token.operator) {
		let categoryType = _getCategoryDataType(options, token.id);

		switch (categoryType) {
			case 'string':
			case 'text':
			case 'email':
				return STRING_OPERATIONS;
			case 'enumoptions':
			case 'boolean':
				return ENUM_OPERATIONS;
			case 'integer':
			case 'float':
			case 'datetime':
			case 'date':
				return NUM_DATE_OPERATIONS;
			case 'async':
				return ASYNC_OPERATIONS;
			default:
				return [];
		}
	} else {
		return _getCategoryOptions(options, token.id);
	}
}


// Get the data type of a category
// defaults to string if an error occurs
export function	_getCategoryDataType(options, id) {
	let category = options.find(option => {
		return option.id === id;
	});

	if (!category) {
		return 'string';
	}

	if (!!category.options) {
		return 'enumoptions';
	} else if (!!category.asyncFilter) {
		return 'async';
	} else {
		return category.type || 'string';
	}
}

// Get the available options(enum) if any were passed in with the 
// options object
export function _getCategoryOptions(options = [], id) {
	let category = options.find(option => {
		return option.id === id;
	});

	if (!category) {
		return [];
	}
	
	// default case for boolean data types
	if (category.type === 'boolean' && !category.options) {
		return ['True', 'False'];
	}

	return category.options;
}

// gets the next header to display over the selectable list of options
export function _getHeader(nextToken = {}) {
	if (nextToken.category === '') {
		return 'Field';
	} else if (nextToken.operator === '') {
		return 'Operator';
	} else {
		return 'Value';
	}
}

// Get the input data type after a user selects a category and operation
// Used to render possible operations on that data
// Renders to string if category and operator have been selected
export function _getInputDatatype(token, options) {
	if (!!token.category && !!token.operator) {
		return _getCategoryDataType(options, token.id);
	}

	return 'string';
}

// Check a token against the current list of tokens for duplicates
export function _isDuplicateToken(tokens, newToken) {
	return tokens.some(token => {
		return token.category === newToken.category &&
			token.operator === newToken.operator &&
			token.value === newToken.value
	}, this);
}
