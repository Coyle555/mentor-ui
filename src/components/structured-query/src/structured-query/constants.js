const EQUALS = 'equals';
const DOES_NOT_EQUAL = 'does not equal';
const CONTAINS = 'contains';
const DOES_NOT_CONTAIN = 'does not contain';
const LESS_THAN = 'less than';
const LESS_THAN_OR_EQUAL_TO = 'less than or equal to';
const GREATER_THAN = 'greater than';
const GREATER_THAN_OR_EQUAL_TO = 'greater than or equal to';
const IS_EMPTY = 'is empty';
const IS_NOT_EMPTY = 'is not empty';

// mapping used to check validity of initial tokens
export const ALL_OPERATIONS = {
	EQUALS,
	DOES_NOT_EQUAL,
	CONTAINS,
	DOES_NOT_CONTAIN,
	LESS_THAN,
	LESS_THAN_OR_EQUAL_TO,
	GREATER_THAN,
	GREATER_THAN_OR_EQUAL_TO,
	IS_EMPTY,
	IS_NOT_EMPTY
};

// operations for strings
export const STRING_OPERATIONS = [
	EQUALS,
	DOES_NOT_EQUAL,
	CONTAINS,
	DOES_NOT_CONTAIN,
	IS_EMPTY,
	IS_NOT_EMPTY
];
// operations for integers and dates
export const NUM_DATE_OPERATIONS = [
	EQUALS,
	DOES_NOT_EQUAL,
	LESS_THAN,
	LESS_THAN_OR_EQUAL_TO,
	GREATER_THAN,
	GREATER_THAN_OR_EQUAL_TO,
	IS_EMPTY,
	IS_NOT_EMPTY
];
// operations for enumerable options
export const ENUM_OPERATIONS = [
	EQUALS,
	DOES_NOT_EQUAL,
	IS_EMPTY,
	IS_NOT_EMPTY
];

export const ASYNC_OPERATIONS = [
	CONTAINS,
	DOES_NOT_CONTAIN,
	IS_EMPTY,
	IS_NOT_EMPTY
];

