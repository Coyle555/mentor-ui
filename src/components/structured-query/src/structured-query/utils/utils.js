import {
	ALL_OPERATIONS,
	ENUM_OPERATIONS,
	NUM_DATE_OPERATIONS,
	STRING_OPERATIONS
} from '../constants';

// make sure a token is valid before the structured filter uses it
export function validateToken(token) {
	if (typeof token !== 'object'
		|| !token.id
		|| !token.label
		|| !token.operator) {

		return false;
	}

	// no valid value when operation isnt a empty check
	if (!token.value
		&& token.operator !== ALL_OPERATIONS.IS_EMPTY
		&& token.operator !== ALL_OPERATIONS.IS_NOT_EMPTY) {

		return false;
	}

	// operation does not exist in list of approved operations
	if (!ALL_OPERATIONS[token.operator.toUpperCase().replace(/ /g, '_')]) {
		return false;
	}

	return true;
}

// Get the options available based on where the user is in the query
export function _getOptionsForTypeahead(fields = [], token = {}) {
	if (!token.label) {

		return fields.map(field => field.label);

	} else if (!token.operator) {
		let labelType = _getLabelDataType(fields, token.id);

		switch (labelType) {
			case 'string':
			case 'text':
			case 'email':
				return STRING_OPERATIONS;
			case 'enumoptions':
			case 'boolean':
				return ENUM_OPERATIONS;
			case 'integer':
			case 'float':
			case 'datetime':
			case 'date':
				return NUM_DATE_OPERATIONS;
			default:
				return [];
		}
	} else {
		return _getLabelOptions(fields, token.id);
	}
}


// Get the data type of a label
// defaults to string if an error occurs
export function	_getLabelDataType(fields = [], id) {
	let label = fields.find(field => field.id === id);

	if (!label) {
		return 'string';
	}

	if (!!label.options) {
		return 'enumoptions';
	} else {
		return label.type || 'string';
	}
}

// Get the available options(enum) if any were passed in with the 
// options object
export function _getLabelOptions(fields = [], id) {
	const label = fields.find(field => field.id === id);

	if (!label) return [];
	
	// default case for boolean data types
	if (label.type === 'boolean' && !label.options) {
		return ['True', 'False'];
	}

	return label.options;
}

// get the parse functions for an options list if it exists
export function _getParseForOptions(fields = [], token) {
	if (fields.length === 0 || !token.id || !token.label || !token.operator) {
		return null;
	}

	const field = fields.find(field => field.id === token.id);
	return field.parse;
}

// gets the next header to display over the selectable list of options
export function _getHeader(nextToken = {}) {
	if (nextToken.label === '') {
		return 'Field';
	} else if (nextToken.operator === '') {
		return 'Operator';
	} else {
		return 'Value';
	}
}

// Get the input data type after a user selects a label and operation
// Used to render possible operations on that data
// Renders to string if label and operator have been selected
export function _getInputDatatype(token, fields) {
	if (!!token.label && !!token.operator) {
		return _getLabelDataType(fields, token.id);
	}

	return 'string';
}

// Check a token against the current list of tokens for duplicates
export function _isDuplicateToken(tokens, newToken, parse) {
	let newTokenValue = typeof parse === 'function'
		? parse(newToken.value)
		: newToken.value;
	let tokenValue;

	return tokens.some(token => {
		tokenValue = typeof parse === 'function'
			? parse(token.value)
			: token.value;

		return token.label === newToken.label &&
			token.operator === newToken.operator &&
			tokenValue === newTokenValue;
	}, this);
}
