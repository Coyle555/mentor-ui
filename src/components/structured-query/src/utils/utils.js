import {
	ALL_OPERATIONS,
	ENUM_OPERATIONS,
	NUM_DATE_OPERATIONS,
	STRING_OPERATIONS
} from '../constants';

// make sure a token is valid before the structured filter uses it
export function validateToken(token) {
	if (typeof token !== 'object'
		|| !token.id
		|| !token.label
		|| !token.operator) {

		return false;
	}

	// no valid value when operation isnt a empty check
	if (!token.value
		&& token.operator !== ALL_OPERATIONS.IS_EMPTY
		&& token.operator !== ALL_OPERATIONS.IS_NOT_EMPTY) {

		return false;
	}

	// operation does not exist in list of approved operations
	if (!ALL_OPERATIONS[token.operator.toUpperCase().replace(/ /g, '_')]) {
		return false;
	}

	return true;
}

// Get the options available based on where the user is in the query
export function _getOptionsForTypeahead(fields = [], token = {}) {
	if (!token.label) {

		return fields.map(field => field.label);

	} else if (!token.operator) {
		let labelType = _getFieldDataType(fields, token.id);

		switch (labelType) {
			case 'enumoptions':
			case 'boolean':
				return ENUM_OPERATIONS;
			case 'integer':
			case 'float':
			case 'datetime':
			case 'date':
				return NUM_DATE_OPERATIONS;
			default:
				return STRING_OPERATIONS;
		}
	} else {
		return _getLabelOptions(fields, token.id);
	}
}


// Get the data type of a label
// defaults to string if an error occurs
export function	_getFieldDataType(fields = [], id) {
	let field = fields.find(field => field.id === id);

	if (!field) {
		return 'string';
	}

	if (Array.isArray(field.options)) {
		return 'enumoptions';
	} else {
		return field.type || 'string';
	}
}

// Get the available options(enum) if any were passed in with the 
// options object
export function _getLabelOptions(fields = [], id) {
	const label = fields.find(field => field.id === id);

	if (!label) return [];
	
	// default case for boolean data types
	if (label.type === 'boolean' && !Array.isArray(label.options)) {
		return ['Yes', 'No'];
	}

	return label.options;
}

// gets the next header to display over the selectable list of options
export function _getHeader(nextToken = {}) {
	if (nextToken.label === '') {
		return 'Field';
	} else if (nextToken.operator === '') {
		return 'Operator';
	} else {
		return 'Value';
	}
}

// Get the input data type after a user selects a label and operation
// Used to render possible operations on that data
// Renders to string if label and operator have been selected
export function _getInputDatatype(token, fields) {
	if (!!token.label && !!token.operator) {
		return _getFieldDataType(fields, token.id);
	}

	return 'string';
}

// Check a token against the current list of tokens for duplicates
export function _isDuplicateToken(tokens, newToken) {
	return tokens.some(token => (
		token.label === newToken.label &&
		token.operator === newToken.operator &&
		token.value === newToken.value
	), this);
}
