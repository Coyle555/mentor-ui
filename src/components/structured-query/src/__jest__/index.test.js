jest.mock('react-tooltip', () => (props) => <div>{JSON.stringify(props)}</div>);

import React from 'react';
import { StructuredQuery } from '../index';
import { keyEvent } from 'utils';
import { ALL_OPERATIONS } from '../constants';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';

afterEach(cleanup);

describe('Render states of the StructuredQuery', () => {
	
	test('Default render', () => {
		const tree = renderer.create(<StructuredQuery />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Custom container class', () => {
		const tree = renderer.create(<StructuredQuery customClasses={{ container: 'foo' }} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Export search callback', () => {
		const tree = renderer.create(<StructuredQuery exportSearch={() => {}} />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Initial tokens in StructuredQuery', () => {

	test('Initial tokens on mount', () => {
		const instance = renderer.create(
			<StructuredQuery initTokens={[
				{ id: 'foo', label: 'Foo', operator: 'equals', value: 'bar' },
				{ id: 'baz', label: 'Baz', operator: 'is empty' }
			]} />
		).getInstance();

		expect(instance.state.searchTokens).toEqual([
			{ id: 'foo', label: 'Foo', operator: 'equals', value: 'bar' },
			{ id: 'baz', label: 'Baz', operator: 'is empty' }
		]);
	});
	
	test('New tokens passed in', () => {
		const render = renderer.create(
			<StructuredQuery initTokens={[
				{ id: 'foo', label: 'Foo', operator: 'equals', value: 'bar' },
			]} />
		);

		render.update(
			<StructuredQuery initTokens={[
				{ id: 'baz', label: 'Baz', operator: 'is empty' }
			]} />
		);

		expect(render.getInstance().state.searchTokens).toEqual([
			{ id: 'baz', label: 'Baz', operator: 'is empty' }
		]);
	});
	
	test('Invalid tokens on mount', () => {
		const instance = renderer.create(
			<StructuredQuery initTokens={[
				{ label: 'Foo', operator: 'equals', value: 'bar' },
				{ id: 'baz', label: 'Baz', operator: 'equals' }
			]} />
		).getInstance();

		expect(instance.state.searchTokens).toEqual([]);
	});
});

describe('Clearing search in the StructuredQuery', () => {

	test('Clear search', () => {
		const onTokenRemove = jest.fn();

		const instance = renderer.create(
			<StructuredQuery
				initTokens={[{
					id: 'foo',
					label: 'Foo',
					operator: 'equals',
					value: 'bar'
				}]}
				onTokenRemove={onTokenRemove}
			/>
		).getInstance();

		instance.clearSearch();
		expect(instance.state.searchTokens).toEqual([]);
		expect(onTokenRemove).toHaveBeenCalledWith([]);
	});
});

describe('Exporting search in the StructuredQuery', () => {

	test('Export search', () => {
		const exportSearch = jest.fn();

		const instance = renderer.create(
			<StructuredQuery
				exportSearch={exportSearch}
				initTokens={[{
					id: 'foo',
					label: 'Foo',
					operator: 'equals',
					value: 'bar'
				}]}
			/>
		).getInstance();

		instance.exportSearch();
		expect(exportSearch).toHaveBeenCalledWith([{
			id: 'foo',
			label: 'Foo',
			operator: 'equals',
			value: 'bar'
		}]);
	});
});

describe('Handling key down events', () => {

	test('A non backspace key down event', () => {
		const instance = renderer.create(<StructuredQuery />).getInstance();

		expect(instance._onKeyDown({ keyCode: 90 })).toBe(undefined);
	});

	test('A key down event with a value existing', () => {
		const instance = renderer.create(<StructuredQuery />).getInstance();

		expect(instance._onKeyDown({ keyCode: keyEvent.DOM_VK_BACK_SPACE }, 'test')).toBe(undefined);
	});

	test('Removing operator using backspace', () => {
		const instance = renderer.create(<StructuredQuery />).getInstance();

		instance.state.nextToken = { id: 'foo', label: 'Foo', operator: 'equals', value: 'foo' };
		instance._onKeyDown({
			keyCode: keyEvent.DOM_VK_BACK_SPACE,
			preventDefault: jest.fn()
		});

		expect(instance.state.nextToken).toEqual({
			id: 'foo',
			label: 'Foo',
			operator: '',
			value: ''
		});
	});

	test('Removing label using backspace', () => {
		const instance = renderer.create(<StructuredQuery />).getInstance();

		instance.state.nextToken = { id: 'foo', label: 'Foo', operator: '', value: '' };
		instance._onKeyDown({
			keyCode: keyEvent.DOM_VK_BACK_SPACE,
			preventDefault: jest.fn()
		});

		expect(instance.state.nextToken).toEqual({
			id: '',
			label: '',
			operator: '',
			type: '',
			value: ''
		});
	});
});

describe('Adding a field to a token', () => {
	const fields = [
		{ id: 'foo', label: 'Foo', type: 'string' },
		{ id: 'bar', label: 'Bar', type: 'string' }
	];
	
	test('Adding a label to a token', () => {
		const instance = renderer.create(<StructuredQuery fields={fields} />).getInstance();

		instance._addTokenForValue('Foo');
		expect(instance.state.nextToken).toEqual({
			id: 'foo',
			label: 'Foo',
			operator: '',
			type: 'string',
			value: ''
		});
	});

	test('Adding an equals operator to a token', () => {
		const instance = renderer.create(<StructuredQuery fields={fields} />).getInstance();

		instance._addTokenForValue('Foo');
		instance._addTokenForValue(ALL_OPERATIONS.EQUALS);
		expect(instance.state.nextToken).toEqual({
			id: 'foo',
			label: 'Foo',
			operator: ALL_OPERATIONS.EQUALS,
			type: 'string',
			value: ''
		});
	});

	test('Adding a is empty operator to a token', () => {
		const onTokenAdd = jest.fn();
		const instance = renderer.create(
			<StructuredQuery onTokenAdd={onTokenAdd} fields={fields} />
		).getInstance();

		instance._addTokenForValue('Foo');
		instance._addTokenForValue(ALL_OPERATIONS.IS_EMPTY);

		expect(instance.state.nextToken).toEqual({
			id: '',
			label: '',
			operator: '',
			type: '',
			value: ''
		});

		expect(instance.state.searchTokens).toEqual([{
			id: 'foo',
			label: 'Foo',
			operator: ALL_OPERATIONS.IS_EMPTY,
			type: 'string',
			value: ''
		}]);

		expect(onTokenAdd).toHaveBeenCalledWith([{
			id: 'foo',
			label: 'Foo',
			operator: ALL_OPERATIONS.IS_EMPTY,
			type: 'string',
			value: ''
		}]);
	});

	test('Adding a is not empty operator to a token', () => {
		const onTokenAdd = jest.fn();
		const instance = renderer.create(
			<StructuredQuery onTokenAdd={onTokenAdd} fields={fields} />
		).getInstance();

		instance._addTokenForValue('Foo');
		instance._addTokenForValue(ALL_OPERATIONS.IS_NOT_EMPTY);

		expect(instance.state.nextToken).toEqual({
			id: '',
			label: '',
			operator: '',
			type: '',
			value: ''
		});

		expect(instance.state.searchTokens).toEqual([{
			id: 'foo',
			label: 'Foo',
			operator: ALL_OPERATIONS.IS_NOT_EMPTY,
			type: 'string',
			value: ''
		}]);

		expect(onTokenAdd).toHaveBeenCalledWith([{
			id: 'foo',
			label: 'Foo',
			operator: ALL_OPERATIONS.IS_NOT_EMPTY,
			type: 'string',
			value: ''
		}]);
	});

	test('Adding a value to a token', () => {
		const onTokenAdd = jest.fn();
		const instance = renderer.create(
			<StructuredQuery onTokenAdd={onTokenAdd} fields={fields} />
		).getInstance();

		instance._addTokenForValue('Foo');
		instance._addTokenForValue(ALL_OPERATIONS.EQUALS);
		instance._addTokenForValue('testVal');

		expect(instance.state.nextToken).toEqual({
			id: '',
			label: '',
			operator: '',
			type: '',
			value: ''
		});

		expect(instance.state.searchTokens).toEqual([{
			id: 'foo',
			label: 'Foo',
			operator: ALL_OPERATIONS.EQUALS,
			type: 'string',
			value: 'testVal'
		}]);

		expect(onTokenAdd).toHaveBeenCalledWith([{
			id: 'foo',
			label: 'Foo',
			operator: ALL_OPERATIONS.EQUALS,
			type: 'string',
			value: 'testVal'
		}]);
	});

	test('Adding a duplicate token', () => {
		const fields = [
			{ id: 'foo', label: 'Foo', type: 'string', options: [{ name: 'foo' }], parse: val => val.name },
			{ id: 'bar', label: 'Bar', type: 'string' }
		];
		const onTokenAdd = jest.fn();
		const instance = renderer.create(
			<StructuredQuery onTokenAdd={onTokenAdd} fields={fields} />
		).getInstance();

		instance._addTokenForValue('Foo');
		instance._addTokenForValue(ALL_OPERATIONS.EQUALS);
		instance._addTokenForValue({ name: 'foo' });

		instance._addTokenForValue('Foo');
		instance._addTokenForValue(ALL_OPERATIONS.EQUALS);
		instance._addTokenForValue({ name: 'foo' });

		expect(instance.state.searchTokens.length).toBe(1);
	});
});

describe('Removing a token', () => {
	const initTokens = [
		{ id: 'foo', label: 'Foo', operator: 'equals', value: 'bar' },
		{ id: 'baz', label: 'Baz', operator: 'is empty' }
	];

	test('Token not found', () => {
		const instance = renderer.create(
			<StructuredQuery initTokens={initTokens} />
		).getInstance();

		expect(instance._removeTokenForValue({})).toBe(undefined);
	});

	test('Token removed', () => {
		const onTokenRemove = jest.fn();

		const instance = renderer.create(
			<StructuredQuery initTokens={initTokens} onTokenRemove={onTokenRemove} />
		).getInstance();

		instance._removeTokenForValue(initTokens[1]);
		expect(instance.state.searchTokens).toEqual([{
			id: 'foo',
			label: 'Foo',
			operator: 'equals',
			value: 'bar'
		}]);

		expect(onTokenRemove).toHaveBeenCalledWith([{
			id: 'foo',
			label: 'Foo',
			operator: 'equals',
			value: 'bar'
		}]);
	});
});
