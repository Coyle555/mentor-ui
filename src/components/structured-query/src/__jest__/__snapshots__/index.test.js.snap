// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render states of the StructuredQuery Custom container class 1`] = `
<div
  className="filter-tokenizer foo"
>
  <span
    className="left-addon first-addon export-search"
    data-for="structured-query-tooltip"
    data-tip="Save Search"
    onClick={[Function]}
  >
    <i
      className="fal fa-save"
    />
  </span>
  <span
    className="left-addon active-filter-container"
    data-for="structured-query-tooltip"
    data-tip="View Filters"
    onClick={[Function]}
  >
    <i
      className="far fa-list"
    />
    <span
      className="active-filter-count"
    >
      0
    </span>
  </span>
  <div
    className="filter-input-group"
  >
    <div
      className="filter-category"
    >
      
    </div>
    <div
      className="filter-operator"
    >
      
    </div>
    <div
      className="typeahead"
    >
      <input
        className=""
        data-testid="typeahead"
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    {"id":"structured-query-tooltip","place":"top","type":"dark","effect":"solid","multiline":false}
  </div>
</div>
`;

exports[`Render states of the StructuredQuery Default render 1`] = `
<div
  className="filter-tokenizer"
>
  <span
    className="left-addon first-addon export-search"
    data-for="structured-query-tooltip"
    data-tip="Save Search"
    onClick={[Function]}
  >
    <i
      className="fal fa-save"
    />
  </span>
  <span
    className="left-addon active-filter-container"
    data-for="structured-query-tooltip"
    data-tip="View Filters"
    onClick={[Function]}
  >
    <i
      className="far fa-list"
    />
    <span
      className="active-filter-count"
    >
      0
    </span>
  </span>
  <div
    className="filter-input-group"
  >
    <div
      className="filter-category"
    >
      
    </div>
    <div
      className="filter-operator"
    >
      
    </div>
    <div
      className="typeahead"
    >
      <input
        className=""
        data-testid="typeahead"
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    {"id":"structured-query-tooltip","place":"top","type":"dark","effect":"solid","multiline":false}
  </div>
</div>
`;

exports[`Render states of the StructuredQuery Export search callback 1`] = `
<div
  className="filter-tokenizer"
>
  <span
    className="left-addon first-addon export-search export-search-enabled"
    data-for="structured-query-tooltip"
    data-tip="Save Search"
    onClick={[Function]}
  >
    <i
      className="fal fa-save"
    />
  </span>
  <span
    className="left-addon active-filter-container"
    data-for="structured-query-tooltip"
    data-tip="View Filters"
    onClick={[Function]}
  >
    <i
      className="far fa-list"
    />
    <span
      className="active-filter-count"
    >
      0
    </span>
  </span>
  <div
    className="filter-input-group"
  >
    <div
      className="filter-category"
    >
      
    </div>
    <div
      className="filter-operator"
    >
      
    </div>
    <div
      className="typeahead"
    >
      <input
        className=""
        data-testid="typeahead"
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    {"id":"structured-query-tooltip","place":"top","type":"dark","effect":"solid","multiline":false}
  </div>
</div>
`;
